assets/js/components/OverlayNotification/OverlayNotificationsRenderer.js

```diff
@@ -24,23 +24,30 @@ import { Fragment } from '@wordpress/element';
 /**
  * Internal dependencies
  */
+import { useSelect } from 'googlesitekit-data';
 import { useFeature } from '../../hooks/useFeature';
 import AudienceSegmentationIntroductoryOverlayNotification from '../../modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationIntroductoryOverlayNotification';
 import AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification from './AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification';
 import LinkAnalyticsAndAdSenseAccountsOverlayNotification from './LinkAnalyticsAndAdSenseAccountsOverlayNotification';
 import { PublicationApprovedOverlayNotification } from '../../modules/reader-revenue-manager/components/dashboard';
+import { MODULES_ANALYTICS_4 } from '../../modules/analytics-4/datastore/constants';
 
 export default function OverlayNotificationsRenderer() {
 	const audienceSegmentationEnabled = useFeature( 'audienceSegmentation' );
+	const audienceSegmentationSetupComplete = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getAudienceSegmentationSetupComplete()
+	);
```

Comment:
This results in the banner being shown for all users, whereas we only want to show it for secondary users i.e. not for the admin who initially set the feature up. We missed a few details in the IB.

In fact we are going to need to make a change to the `getAudienceSegmentationSetupComplete()` setting so that instead of simply storing a boolean, it should store the ID of the user who set the feature up, so we can then check here if the ID is a match for the current user. It's a bit of a pain but we should rename `getAudienceSegmentationSetupComplete()` to `getAudienceSegmentationSetupCompletedBy()` at the same time, to make it semantically clear.

So, please can you apply these changes to the setting, but also rename it to `getAudienceSegmentationSetupCompletedBy()`. You'll need to update a few tests as a result, it's unfortunate but we do need to do it here.

```diff
diff --git a/assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js b/assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js
index f7f41a78b0..4d8a407aa8 100644
--- a/assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js
+++ b/assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js
@@ -193,7 +193,7 @@ function AudienceSegmentationSetupCTAWidget( { Widget, WidgetNull } ) {
 	}
 
 	if (
-		audienceSegmentationSetupComplete ||
+		audienceSegmentationSetupComplete !== null ||
 		configuredAudiences === undefined ||
 		configuredAudiences?.length ||
 		! analyticsIsDataAvailableOnLoad ||
diff --git a/assets/js/modules/analytics-4/datastore/audiences.js b/assets/js/modules/analytics-4/datastore/audiences.js
index a910222a32..2c40456f1a 100644
--- a/assets/js/modules/analytics-4/datastore/audiences.js
+++ b/assets/js/modules/analytics-4/datastore/audiences.js
@@ -522,8 +522,10 @@ const baseActions = {
 			return { error };
 		}
 
+		const userID = select( CORE_USER ).getID();
+
 		dispatch( MODULES_ANALYTICS_4 ).setAudienceSegmentationSetupComplete(
-			true
+			userID
 		);
 
 		const { saveSettingsError } = yield commonActions.await(
diff --git a/assets/js/modules/analytics-4/index.js b/assets/js/modules/analytics-4/index.js
index 42833473f1..a7b5098c1a 100644
--- a/assets/js/modules/analytics-4/index.js
+++ b/assets/js/modules/analytics-4/index.js
@@ -175,7 +175,7 @@ export const registerWidgets = ( widgets ) => {
 				return (
 					availableAudiences?.length &&
 					configuredAudiences === null &&
-					audienceSegmentationSetupComplete === true
+					audienceSegmentationSetupComplete !== null
 				);
 			},
 		},
diff --git a/includes/Modules/Analytics_4/Settings.php b/includes/Modules/Analytics_4/Settings.php
index 1531a67c77..3c74f3d8a4 100644
--- a/includes/Modules/Analytics_4/Settings.php
+++ b/includes/Modules/Analytics_4/Settings.php
@@ -107,7 +107,7 @@ class Settings extends Module_Settings implements Setting_With_Owned_Keys_Interf
 			'adsLinkedLastSyncedAt'             => 0,
 			'availableAudiences'                => null,
 			'availableAudiencesLastSyncedAt'    => 0,
-			'audienceSegmentationSetupComplete' => false,
+			'audienceSegmentationSetupComplete' => null,
 			'detectedEvents'                    => array(),
 		);
 	}
@@ -208,7 +208,9 @@ class Settings extends Module_Settings implements Setting_With_Owned_Keys_Interf
 				}
 
 				if ( isset( $option['audienceSegmentationSetupComplete'] ) ) {
-					$option['audienceSegmentationSetupComplete'] = (bool) $option['audienceSegmentationSetupComplete'];
+					if ( ! is_int( $option['audienceSegmentationSetupComplete'] ) ) {
+						$option['audienceSegmentationSetupComplete'] = null;
+					}
 				}
 			}
```

Once the above is done, you should update this condition as follows. However one more thing to consider is that it's starting to add quite a bit of logic specific to the `AudienceSegmentationIntroductoryOverlayNotification` component into `OverlayNotificationsRenderer`. So it might be preferable to move this condition into `AudienceSegmentationIntroductoryOverlayNotification` for cleaner encapsulation of its logic. That being the case you could remove the tests you've added for `OverlayNotificationsRenderer`.

```suggestion
	const shouldShowAudienceSegmentationIntroductoryOverlay = useSelect(
		( select ) =>
			select( CORE_MODULES ).isModuleActive( 'analytics-4' ) &&
			( ! isViewOnly ||
				select( CORE_USER ).canViewSharedModule( 'analytics-4' ) ) &&
			select(
				MODULES_ANALYTICS_4
			).getAudienceSegmentationSetupCompletedBy() !==
				select( CORE_USER ).getID()
	);
```